interfaces/acados_template/acados_template/c_templates_tera/Makefile.in

```diff
@@ -191,13 +209,15 @@ CASADI_COST_R_SOURCE=
 CASADI_COST_R_E_SOURCE=
 {% if cost_type == "NONLINEAR_LS" %}
 CASADI_COST_R_SOURCE+= {{ model.name }}_r_cost.c 
-{% endif %}
+{% endif %}casadi_fun:
 {% if cost_type_e == "NONLINEAR_LS" %}
 CASADI_COST_R_E_SOURCE+= {{ model.name }}_r_e_cost.c 
 {% endif %}
 
+
+
 casadi_fun:
-	( cd {{ model.name }}_model; gcc $(ACADOS_FLAGS) -c  $(CASADI_MODEL_SOURCE))
+	( cd {{ model.name }}_model; gcc $(ACADOS_FLAGS) -c  $(CASADI_MODEL_SOURCE)) 
```

Comment:
would have guessed that the flag has to be added here too.
This is not true?

Reply:
After several tests, it appeared that no... 
It looks like it's only when the compiler creates the .so (ocp_shared_lib and sim_shared_lib) that it needs the linker flag.